libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java

```diff
@@ -511,22 +518,23 @@ public void setMinPasscodeLength(int minPasscodeLength) {
              * triggered later from OAuthWebviewHelper.
              */
             if (hasStoredPasscode(SalesforceSDKManager.getInstance().getAppContext())) {
-        		showLockActivity(SalesforceSDKManager.getInstance().getAppContext(),
-        				true);
+                setEnabled(true);
+                showLockActivity(ctx, true);
+                return true;
```

Comment:
great catch, you're right, storeMobilePolicy is not being executed in this case, I'll fix that. 

`minPasscodeLength` was still being updated (looks like it's updated twice), maybe I can clean that up too.